### DISCUSSION TITLE
how get spring security context in resolvers / data loaders

### DISCUSSION THREAD

**Discussion by rullyalves** [ExpediaGroup/graphql-kotlin#1467] [General] [answered]
First, thanks for the great work

I would like to know how to integrate spring server with spring security, having method level security and security context recovery.

**Comment by dariuszkuc** [comment, marked as answer]
Hello 👋 
Starting with the v6, `graphql-kotlin` will automatically propagate reactor context to the resolvers so you would integrate in the same way as any other webservice.

In older versions you will need to do a manual context propagation as described in https://github.com/ExpediaGroup/graphql-kotlin/discussions/1300.

**Reply by rullyalves** [reply]
thanks a lot for the answer, in that case should i access the Principal via ReactiveSecurityContextHolder ?

**Reply by dariuszkuc** [reply]
I believe you may need to first configure `SecurityWebFilterChain` but yes once it is configured then you should be able to get it from the security context holder.
